pkg/provider/azure.go

```diff
@@ -127,13 +181,26 @@ func mkAzure(conf config.Config) (prov *AzureProvider, err error) {
 	return
 }
 
-func azureFromManifest(man *manifest.ProjectManifest) (*AzureProvider, error) {
-	return &AzureProvider{man.Cluster, man.Project, man.Bucket, man.Region, man.Context, nil}, nil
+func AzureFromManifest(man *manifest.ProjectManifest, clientSet *ClientSet) (*AzureProvider, error) {
+	var err error
+	clients := clientSet
+	if clientSet == nil {
+		clients, err = GetClientSet(utils.ToString(man.Context["SubscriptionId"]))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &AzureProvider{man.Cluster, man.Project, man.Bucket, man.Region, man.Context, nil, clients}, nil
```

Comment:
I'm not sure if we need to initialize the client here, since we also don't initialize the writer.

Reply:
we have to initialize the clients here for two reasons:
 - to initialize fake clients for the tests (AzureProvider has private fields and can be done in the same package) and it's the main reason
 - previously the clients were created on demand before the call, now after refactor they must exist when we initialize the Azure provider

Reply:
Now that I think about it I believe I did something similar with the AWS provider a while back.